src/infer_request.cc

```diff
@@ -1182,28 +1201,26 @@ InferenceRequest::Normalize()
     {
       const auto& data_type = input.DType();
 
-      // FIXME: Skip byte size validation for TensorRT backend because it breaks
-      // shape-size assumption. See DLIS-6805 for proper fix for TRT backend
-      // reformat_free tensors.
-      bool skip_byte_size_check = false;
-      constexpr char trt_prefix[] = "tensorrt_";
-      const std::string& platform = model_raw_->Config().platform();
-      skip_byte_size_check |= (platform.rfind(trt_prefix) == 0);
-
-      if (!skip_byte_size_check) {
+      // Non-linear IO format input byte size validation will be handled in the
+      // backend.
+      if (!input.IsNonLinearFormatIo()) {
         TRITONSERVER_MemoryType input_memory_type;
         // Because Triton expects STRING type to be in special format
         // (prepend 4 bytes to specify string length), so need to add all the
         // first 4 bytes for each element to find expected byte size
         if (data_type == inference::DataType::TYPE_STRING) {
           RETURN_IF_ERROR(
               ValidateBytesInputs(input_id, input, &input_memory_type));
+
           // FIXME: Temporarily skips byte size checks for GPU tensors. See
           // DLIS-6820.
-          skip_byte_size_check |=
-              (input_memory_type == TRITONSERVER_MEMORY_GPU);
         } else {
-          const auto& input_dims = input.ShapeWithBatchDim();
+          // Shape tensor with dynamic batching does not introduce a new
```

Comment:
Can this be handled within ShapeWIthBatchDim call itself?
As in the input object already knows whether it is a shape tensor or not. How about not adding an extra batch dim in such a scenario.

Reply:
@tanmayv25, I made the change above but had to undo it because we encountered an error when using the model config with batch size enabled ([shape mismatch due to batch size not being included](https://github.com/triton-inference-server/tensorrt_backend/blob/main/src/tensorrt_utils.cc#L374C3-L382C4)). We also may encounter issues in other areas while using `ShapeWithBatchDim()` for shape tensors. Can we proceed with the current change?
Thank you.